google_guest_agent/metadata.go

```diff
@@ -230,5 +233,32 @@ func getMetadata(ctx context.Context, hang bool) (*metadata, error) {
 		return nil, err
 	}
 	var ret metadata
-	return &ret, json.Unmarshal(md, &ret)
+	if err := json.Unmarshal(md, &ret); err != nil {
+		return nil, jsonError(md, err)
+	}
+	return &ret, nil
+}
+
+func jsonError(data []byte, err error) error {
```

Comment:
it seems like this should be incorporated into our unmarshal rather than invoked optionally when handling errors. but i like the function!

Reply:
This was the quickest way to get this function in to log errors, its a modification of what I do with daisy

Reply:
can you add TODO if it's too much work to put this in return of Unmarshal() ?